ledger/store/src/block/mod.rs

```diff
@@ -1360,6 +1360,11 @@ impl<N: Network, B: BlockStorage<N>> BlockStore<N, B> {
         self.storage.id_map().keys_confirmed()
     }
 
+    /// Returns the height of the latest block in the storage.
+    pub fn max_height(&self) -> Option<u32> {
+        u32::try_from(self.storage.id_map().len_confirmed()).ok()?.checked_sub(1)
```

Comment:
in order to make sure this matches the current logic, there are 2 scenarios to consider:
1. empty ledger: in that case `len_confirmed == 0` (there are no records mapping block height to hash), and `checked_sub(1)` results in `None` being returned (which is equivalent to the current `heights().max()` for an empty `id_map`)
2. `N` blocks (including `1` for just the genesis): `len_confirmed == N`, and `checked_sub(1)` returns `Some(N - 1)`, which is the same as max block height (since they are all unique and consecutive)

in addition, `u32::try_from` will never fail here, as the block height can not have a value greater than `u32::MAX`